### PR TITLE
xblock-image-explorer | Drop py3.8 support | replace pkg_resource with importlib.resources

### PR DESCRIPTION
**Pull request by OmerBhatti**
- Drop Py3.8 support
 - Replace pkg_resources lib with importlib.resources

**Ticket:** [Move on from deprecated pkg_resources api #150](https://github.com/openedx/xblock-image-explorer/issues/150)

<mark>PR is expected to merge after merging this PR : [chore: Upgrade Python requirements](https://github.com/openedx/xblock-image-explorer/pull/151)</mark>

Followed migration guide: https://importlib-resources.readthedocs.io/en/latest/migration.html

**Testing:**
I have tested the changes with following steps:

 - Install xblock-image-explorer into xblock-sdk
 - Test image explorer xblock functionality
 - Run tox tests

## Screenshots
image

Resolves https://github.com/openedx/xblock-image-explorer/issues/159.

### CODE REVIEW

**Comment by mphilbrick211** [comment]
Hi @OmerBhatti! Thanks for this contribution!

If you are contributing on behalf of edly, please have your manager reach out to oscm@axim.org to have you added to our existing entity agreement with edly. Thank you!

**Comment by farhan** [comment]
@feanil @kdmccormick 
Can you guys please merge [this](https://github.com/openedx/xblock-image-explorer/pull/155) as its blocking this PR. Also remove Python 3.8 from settings

**Comment by farhan** [comment]
@OmerBhatti Once the Python dependencies PR merges, rebase this PR branch, resolve conflicts if required.

**Review by farhan — CHANGES_REQUESTED** [review]

File: image_explorer/image_explorer.py
Comment (farhan) [inline image_explorer/image_explorer.py:467]: We can simply return

```python
return loader.load_unicode(path)
```

**Review by Agrendalath — CHANGES_REQUESTED** [review]
We're dropping Python 3.8 support here, so we should make the major version bump in `image_explorer/__init__.py` and indicate in commit messages that this is a breaking change.

**Comment by e0d** [comment]
@Agrendalath looks like your comment has been addressed, can you resolve?

File: .github/workflows/ci.yml
Comment (Agrendalath) [inline .github/workflows/ci.yml:21]: Why are we reverting the os version?
https://github.com/openedx/public-engineering/issues/278

**Review by Agrendalath — CHANGES_REQUESTED** [review]
@e0d, I don't see any changes mentioned in https://github.com/openedx/xblock-image-explorer/pull/153#pullrequestreview-2292142508 in this PR. Am I missing something?

File: .github/workflows/ci.yml
Comment (e0d) [reply .github/workflows/ci.yml:21]: Agree with the Ubuntu comment, but isn't the line below what you were requesting vis-a-via python?

`python-version: ['3.11', '3.12']`

Comment (Agrendalath) [reply .github/workflows/ci.yml:21]: @e0d, I'm afraid I don't understand this message. Since we are dropping Python 3.8 support here, we should update the version of this XBlock (in `image_explorer/__init__.py`) and [mark it as a breaking change](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html) because this XBlock will no longer work with pre-Redwood releases.

Comment (OmerBhatti) [reply .github/workflows/ci.yml:21]: it might've happen mistakenly while rebasing. i'll revert it back

**Review by Agrendalath — APPROVED** [review]
:+1:

* [x] I tested this: checked that the CI is passing; I didn't test it with Tutor
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a

**Review by farhan — APPROVED** [review]

**Pull request merged**
